lib/api/model/model.dart

```diff
@@ -531,10 +532,143 @@ String? tryParseEmojiCodeToUnicode(String emojiCode) {
   }
 }
 
+/// The name of a Zulip topic.
+// TODO(dart): Can we forbid calling Object members on this extension type?
+//   (The lack of "implements Object" ought to do that, but doesn't.)
+//   In particular an interpolation "foo > $topic" is a bug we'd like to catch.
+// TODO(dart): Can we forbid using this extension type as a key in a Map?
+//   (The lack of "implements Object" arguably should do that, but doesn't.)
+//   Using as a Map key is almost certainly a bug because it won't case-fold;
+//   see for example #739, #980, #1205.
+extension type const TopicName(String _value) {
```

Comment:
nit: how about pulling out this move as a separate prep commit? I think that'd make the Conversation commit a bit easier to read.

(In particular the three Conversation classes would then appear in the same screenful as the changes to at least StreamMessage, rather than have the old copy of TopicName appear in between.)